quipucords/api/source/serializer.py

```diff
@@ -123,14 +97,16 @@ def create(self, validated_data):
         source_type = validated_data.get('source_type')
         credentials = validated_data.pop('credentials')
         hosts_data = validated_data.pop('hosts', None)
+        if hosts_data:
+            hosts_list = json.loads(hosts_data)
```

Comment:
I think you are assuming that this value will be a list. I think we need to do validation, where we are checking if `isinstance(data, list)` and throw a validation error if the format is wrong. Also need to check that each of the items in the list is a string.

Reply:
check not needed here since it is done in validate_hosts